jsk_naoqi_robot/jsk_pepper_startup/nodes/tweet-server.l

```diff
@@ -0,0 +1,115 @@
+    (send res :data)
+    ))
+
+(defun speak (str)
+  (let ((speech_msg (instance std_msgs::String :init)))
+    (send speech_msg :data str)
+    (ros::publish "/speech" speech_msg))
+  )
+
+(defun take-pose (position_list)
+  (let ((joint-trajectory-action (instance ros::simple-action-client :init
+					   "/pepper_robot/pose/joint_trajectory" naoqi_bridge_msgs::JointTrajectoryAction
+					   ))
+	goal
+	trajectory)
+    (if (send joint-trajectory-action :wait-for-server 3)
+	(progn
+	  (setq goal (send joint-trajectory-action :make-goal-instance))
+	  (setq trajectory (send goal :goal :trajectory))
+	  (send trajectory :joint_names (list "KneePitch" "HipRoll" "HipPitch" "LShoulderPitch" "LShoulderRoll" "LElbowYaw" "LElbowRoll" "LWristYaw" "RShoulderPitch" "RShoulderRoll" "RElbowYaw" "RElbowRoll" "RWristYaw" "HeadYaw" "HeadPitch"))
+	  (send trajectory :points
+		(list (instance trajectory_msgs::JointTrajectoryPoint
+				:init
+				:positions position_list
+				:time_from_start (ros::time 1.5))))
+	  (send joint-trajectory-action :send-goal goal)
+	  (send joint-trajectory-action :wait-for-result)
+	  (send joint-trajectory-action :get-result))
+      (ros::ros-warn "No joint trajectory action found")
+      )))
+
+(defun move-rhand (value)
+  (let ((move_hand_msg (instance naoqi_bridge_msgs::JointAnglesWithSpeed :init)))
+    (send move_hand_msg :header :stamp (ros::time-now))
+    (send move_hand_msg :header :seq 1)
+    (send move_hand_msg :speed 0.5)
+    (send move_hand_msg :relative 0)
+    (send move_hand_msg :joint_names (list "RHand"))
+    (send move_hand_msg :joint_angles (list value))
+    (ros::publish "/pepper_robot/pose/joint_angles" move_hand_msg)))
+
+(defun tweet (str)
+  (let (speech-language (timestamp0 0) (timestamp1 0) (filename "/tmp/pepper_camera.png") tm)
+    (setq speech-language (get-language))
+    ;; taking picture pose
+    (take-pose (list (deg2rad 2.0) (deg2rad -2.0) (deg2rad -5.0) (deg2rad 100.0) (deg2rad 50.0) (deg2rad -20.0) (deg2rad -80.0) (deg2rad -40.0) (deg2rad -60.0) (deg2rad -20.0) (deg2rad -10.0) (deg2rad 60.0) (deg2rad 40.0) 0.0 0.0))
+    (if (string= speech-language "Japanese")
+	(progn
+	  (speak "\\vct=120\\写真を撮るよぉー")
+	  (speak "\\vct=120\\さんっ")
+	  (unix:sleep 2)
+	  (speak "\\vct=120\\にっ")
+	  (unix:sleep 2)
+	  (speak "\\vct=120\\いちっ")
+	  (unix:sleep 2)
+	  (speak"\\vct=120\\カしゃっ！")
+	  )
+      (progn
+	(speak "\\rspd=90\\\\vct=120\\I'll take a picture")
+	(unix:sleep 2)
+	(speak "\\rspd=90\\\\vct=120\\three")
+	(unix:sleep 2)
+	(speak "\\rspd=90\\\\vct=120\\two")
+	(unix:sleep 2)
+	(speak "\\rspd=90\\\\vct=120\\one")
+	(unix:sleep 2)
+	(speak "\\rspd=90\\\\vct=120\\ka sha")
+	))
+    ;; close hand
+    (move-rhand 0.0)
+    (unix:sleep 1)
+    ;; take reset pose
+    (take-pose (list (deg2rad 2.0) (deg2rad -2.0) (deg2rad -5.0) (deg2rad 85.0) (deg2rad 10.0) (deg2rad -70.0) (deg2rad -20.0) (deg2rad -40.0) (deg2rad 85.0) (deg2rad -10.0) (deg2rad 70.0) (deg2rad 20.0) (deg2rad 40.0) 0.0 0.0))
```

Comment:
`map`の使い方が今は分からなくて，分かったら直す！

Reply:
`mapcar #'deg2rad (list 2.0 -2.0 -5.0 85.0 10.0 -70.0 -20.0 -40.0 85.0 -10.0 70.0 20.0 40.0)`

Reply:
If you can use `*pepper*`: `(map float-vector #'deg2rad (send *pepper* :reset-pose))`

Reply:
Thank you for your comment!

I sent PR which just added sample use of `map` and `mapcar` https://github.com/euslisp/EusLisp/pull/382.

memo1:
(`map cons #'deg2rad (list 2.0 -2.0 -5.0 85.0 10.0 -70.0 -20.0 -40.0 85.0 -10.0 70.0 20.0 40.0 0.0 0.0))` also worked :)

memo2:
`dolist` explanation @ j-manual says:
> dolist は、他の繰返しより早く実行される。たとえば、 mapcar や再帰的関数のようなものより。それ
は、 dolist が関数の closure を作ったり適用したりする必要がなく、新しいパラメータのバインドが必要
でないため

If I use `dolist` instead of `mapcar`->
```
(setq angle-list (list 2.0 -2.0 -5.0 85.0 10.0 -70.0 -20.0 -40.0 85.0 -10.0 70.0 20.0 40.0 0.0 0.0))
(setq tmp nil)
(dolist (i angle-list) (setq tmp (append tmp (list (deg2rad i))))) 
```
=> `tmp`: list